src/core/layout/qgslayoutitempage.cpp

```diff
@@ -103,6 +103,47 @@ bool QgsLayoutItemPage::setPageSize( const QString &size, Orientation orientatio
   }
 }
 
+QPageLayout QgsLayoutItemPage::pageLayout() const
+{
+  QPageLayout pageLayout;
+  pageLayout.setMargins( {0, 0, 0, 0} );
+  pageLayout.setMode( QPageLayout::FullPageMode );
+  double factor = 1.;
+
+  switch ( pageSize().units() )
```

Comment:
You could use

    const QSizeF size = layout()->measurementConverter()->convert( pageSize(), QgsUnitTypes::LayoutMillimeters ).toQSizeF();

instead